File: framework/module_service/src/main/java/com/xxl/hello/service/upload/api/UploadService.java
package com.xxl.hello.service.upload.api;

import androidx.annotation.NonNull;

/**
 * 上传服务
 *
 * @author xxl.
 * @date 2022/5/27.
 */
public interface UploadService {

    /**
     * 开始上传
     *
     * @param key
     */
    void onStart(@NonNull final String key);

    /**
     * 上传完成
     *
     * @param domain
     * @param key
     */
    void onComplete(@NonNull final String domain,
                    @NonNull final String key);
}

File: framework/module_service/src/main/java/com/xxl/hello/service/upload/api/UploadService.kt
package com.xxl.hello.service.upload.api

import com.xxl.core.service.upload.UploadListener
import java.io.File

/**
 *
 * @author xxl.
 * @date 2022/5/31.
 */
interface UploadService {

    /**
     * 上传
     * @param file
     * @param listener
     */
    fun upload(file: File,
               listener: UploadListener)
}